File: easy/217-contains-duplicate.ts
// input: nums (int[])
// if value appears <=2, return true; else return false

function containsDuplicate(nums: number[]): boolean {
  // const cache: any = {};
  // for(let i = 0; i < nums.length; i++){
  //   if(!cache[nums[i]]) cache[nums[i]] = 1;
  //   else return true;
  // }
  // return false;

  const set = new Set<number>(nums);
  return set.size < nums.length;
};

export default containsDuplicate;

File: python/217-contains-duplicate.py
from typing import List

class Solution:
  def containsDuplicate(self, nums: List[int]) -> bool:
    cache = set()
    for num in nums:
      if num in cache: return True
      cache.add(num)
    return False

print(Solution().containsDuplicate([1,2,3,1]))